applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nonlinear_strain_energy_response_function.cpp

```diff
@@ -0,0 +1,378 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Mahmoud Sesa, https://github.com/mahmoudsesa
+//
+
+// System includes
+
+// External includes
+#include "utilities/variable_utils.h"
+
+// Project includes
+#include "adjoint_nonlinear_strain_energy_response_function.h"
+
+namespace Kratos 
+{
+    AdjointNonlinearStrainEnergyResponseFunction::AdjointNonlinearStrainEnergyResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings)
+    : AdjointStructuralResponseFunction(rModelPart, ResponseSettings)
+    {
+        // initializing mConditionsRHS 
+        for(auto& cond_i : rModelPart.Conditions())
+        {
+            SizeType number_of_nodes = cond_i.GetGeometry().size();
+            SizeType dimension = cond_i.GetGeometry().WorkingSpaceDimension();
+            SizeType vec_size = number_of_nodes * dimension;
+          
+            mConditionsRHS[cond_i.Id()] = ZeroVector(vec_size);
```

Comment:
I will keep pointing it out, relying on Ids is a weak mechanicsm ;)

Reply:
;-) i believe you, but is there a persistent index/tag for objects like nodes and conditions that does not change after they are created?